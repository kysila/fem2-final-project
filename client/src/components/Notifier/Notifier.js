/* eslint-disable react/prop-types */
import React, { useEffect } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { withSnackbar } from 'notistack';
import CloseIcon from '@material-ui/core/SvgIcon/SvgIcon';
import { closeSnackbar, removeSnackbar } from '../../store/notification/actions';

function Notifier(props) {
  let displayed = [];

  const storeDisplayed = (id) => {
    displayed = [...displayed, id];
  };

  const removeDisplayed = (id) => {
    displayed = displayed.filter((key) => id !== key);
  };

  useEffect(() => {
    const { notifications = [] } = props;

    notifications.forEach(({
      key,
      message,
      options = {},
      dismissed = false,
    }) => {
      if (dismissed) {
        props.closeSnackbar(key);
        return;
      }
      if (displayed.includes(key)) return;
      props.enqueueSnackbar(message, {
        key,
        ...options,
        action: (k) => (
          <CloseIcon style={{ cursor: 'pointer' }} onClick={() => closeSnackbar(k)} />
        ),
        onClose: (event, reason, k) => {
          if (options.onClose) {
            options.onClose(event, reason, k);
          }
        },
        onExited: (event, k) => {
          props.removeSnackbar(k);
          removeDisplayed(k);
        },
      });
      storeDisplayed(key);
    });
  });

  return null;
}

const mapStateToProps = (store) => ({
  notifications: store.notification,
});

const mapDispatchToProps = (dispatch) => bindActionCreators({
  closeSnackbar,
  removeSnackbar,
}, dispatch);

const NotifierConnect = withSnackbar(connect(
  mapStateToProps,
  mapDispatchToProps,
)(Notifier));

export {
  NotifierConnect as Notifier,
};
