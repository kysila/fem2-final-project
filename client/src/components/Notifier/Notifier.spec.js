/* eslint-disable */
import React from 'react';
import {shallow} from 'enzyme';
import {Notifier} from './Notifier';

describe('Notifier component', () => {
  it('Notifier render', () => {
    const rendered = shallow(
      <Notifier />,
    );

    expect(rendered).toMatchSnapshot();
  });
});
