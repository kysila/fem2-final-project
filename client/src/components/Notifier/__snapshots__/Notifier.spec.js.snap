// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notifier component Notifier render 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
